src/pj_fwd.c

```diff
@@ -174,11 +174,23 @@ static PJ_COORD fwd_finalize (PJ *P, PJ_COORD coo) {
 }
 
 
+static PJ_COORD error_or_coord(PJ *P, PJ_COORD coord, int last_errno) {
+    if (proj_errno(P))
+        coord = proj_coord_error();
+    else
+        proj_errno_restore(P, last_errno);
+
+    return coord;
+}
+
 
 XY pj_fwd(LP lp, PJ *P) {
+    int last_errno;
     PJ_COORD coo = {{0,0,0,0}};
     coo.lp = lp;
```

Comment:
It is tempting, at this stage, to consider whether we want to play ball *at all* if there's an unhandled exception - i.e. just do

    if (proj_errno(P))
        return proj_coord_error();

It will, however, require some additional error handling in the `pj_transform()`, and the `proj_trans...()`family, to avoid an error for a single point (e.g. out of area) to result in all subsequent points in that call failing.

(this comment also applies to `pj_fwd3d`, `pj_fwd4d`, and their `inv`counterparts).

Reply:
That's a problem for another day :-)

Reply:
:-)